src/Plugins/Cache.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Peck\Plugins;
+
+final readonly class Cache
+{
+    public function __construct(
+        private string $cacheDirectory,
+    ) {
+        if (! is_dir($this->cacheDirectory)) {
+            mkdir($this->cacheDirectory, 0777, true);
```

Comment:
Of course we could create a custom exception for this, but for now this should be enough

```suggestion
            throw new \RuntimeException("Could not create cache directory: {$this->cacheDirectory}");
```